nio/client/async_client.py

```diff
@@ -1388,6 +1387,7 @@ async def sync_forever(
                     task.cancel()
 
                 raise
+        self._stop_sync_forever = False
```

Comment:
What's the motivation for the change in location? Not opposed to this at all, just wondering what issue this solves

Reply:
Working on a project with [simplematrixbotlib](https://gitlab.com/mb-saces/parametricspace/) we patched the bot-lib to be used as app framework too.

We hit two issues: the app does not want to sync et al (calling client.stop_sync_forever() at startup) or thanks to async and threading, the decision to stop syncing can comes fast, before the regular client.sync_forever() is called.

This patch solves it. Will also add some doc about the exact behaviour.

Reply:
Ahhh, yeah, I can't imagine the behavior was thread-safe before; makes sense to me